system_tests/bigquery.py

```diff
@@ -90,7 +92,15 @@ def test_update_dataset(self):
         after = [grant for grant in dataset.access_grants
                  if grant.entity_id != 'projectWriters']
         dataset.access_grants = after
-        dataset.update()
+
+        # We need to wait for the changes in the dataset to propgate and be
+        # eventually consistent. The alternative outcome is a 403 Forbidden
+        # response from upstream.
```

Comment:
1.  s/propgate/propagate
1.  Eventual consistency is not the error, it's rate-limiting from the API. Instead say

    > BigQuery API uses 403 Forbidden instead of 429 Too Many Requests
    > for rate-limiting. We catch and retry once after a 30s break in that case.